### PR TITLE
[vendor] Verilator DPI memutil

### PR DESCRIPTION
**Pull request by marnovandermaas**
This change vendors in the Verilator DPI fix that was merged into OpenTitan in this PR: https://github.com/lowRISC/opentitan/pull/19112
It is one step in improving the following PR: https://github.com/lowRISC/ibex-demo-system/pull/51

It also pulls in the rest of the updated lowRISC IP, which required an updated patch and DV file list.

### CODE REVIEW

**Comment by nbdd0121** [comment]
Needs rebasing on master?

**Comment by nbdd0121** [comment]
I have check all current 4 commits and they LGTM.

**Comment by marnovandermaas** [comment]
Thanks @nbdd0121, I've made the changes to `common_project_cfg.hjson`. This was due to OpenTitan commit 0bda971

File: dv/uvm/common_project_cfg.hjson [hunk 1]
```diff
@@ -14,6 +14,10 @@
   scratch_path:       "{scratch_base_path}/{branch}"
   tool_srcs_dir:      "{scratch_path}/{tool}"
 
+  //Common DVSIM data structures
```
Comment (nbdd0121) [inline dv/uvm/common_project_cfg.hjson:17]: Missing a space here

**Review by nbdd0121 — APPROVED** [review]
LGTM with missing space added

**Review by hcallahan-lowrisc — APPROVED** [review]

**Review by GregAC — APPROVED** [review]

File: dv/uvm/common_project_cfg.hjson (hunk 1 above)
Comment (marnovandermaas) [reply dv/uvm/common_project_cfg.hjson]: Thanks for pointing that out, this should now be fixed.

**Pull request merged**